File: CS/Q149895/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;

namespace Q149895 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void OnGridControlEmbeddedNavigatorButtonClick(object sender, NavigatorButtonClickEventArgs e) {
            if (e.Button.ButtonType == NavigatorButtonType.Remove) {
                ((XPBaseObject)((GridView)gridControl1.FocusedView).GetFocusedRow()).Delete();
                e.Handled = true;
            }
        }
    }
}

File: VB/Q149895/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.Xpo
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid

Namespace Q149895

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub OnGridControlEmbeddedNavigatorButtonClick(ByVal sender As Object, ByVal e As NavigatorButtonClickEventArgs)
            If e.Button.ButtonType = NavigatorButtonType.Remove Then
                CType(CType(gridControl1.FocusedView, GridView).GetFocusedRow(), XPBaseObject).Delete()
                e.Handled = True
            End If
        End Sub
    End Class
End Namespace
